lms/djangoapps/completion/tests/test_handlers.py

```diff
@@ -36,51 +38,105 @@ def override_waffle_switch(self, override):
         self.addCleanup(_waffle_overrider.__exit__, None, None, None)
 
 
+class CustomScorableBlock(XBlock):
+    """
+    A scorable block with a custom completion strategy.
+    """
+    has_score = True
+    has_custom_completion = True
+    completion_mode = XBlockCompletionMode.COMPLETABLE
+
+
+class ExcludedScorableBlock(XBlock):
+    """
+    A scorable block that is excluded from completion tracking.
+    """
+    has_score = True
+    has_custom_completion = False
+    completion_mode = XBlockCompletionMode.EXCLUDED
+
+
 @ddt.ddt
-class ScorableCompletionHandlerTestCase(CompletionHandlerMixin, TestCase):
+class ScorableCompletionHandlerTestCase(CompletionWaffleMixin, TestCase):
     """
     Test the signal handler
     """
 
     def setUp(self):
         super(ScorableCompletionHandlerTestCase, self).setUp()
+        self.course_key = CourseKey.from_string('edx/course/beta')
+        self.scorable_block_key = self.course_key.make_usage_key(block_type='problem', block_id='red')
         self.user = UserFactory.create()
-        self.course_key = CourseKey.from_string("course-v1:a+valid+course")
-        self.block_key = self.course_key.make_usage_key(block_type="video", block_id="mah-video")
         self.override_waffle_switch(True)
 
-    @ddt.data(
-        ({'score_deleted': True}, 0.0),
-        ({'score_deleted': False}, 1.0),
-        ({}, 1.0),
-    )
-    @ddt.unpack
-    def test_handler_submits_completion(self, params, expected_completion):
+    def call_handler_for_block(self, block_key, score_deleted=None):
+        """
+        Call the signal handler for the specified block.
```

Comment:
Done.  I left it truncated in the test names because they will show up in the test report appended to the class name, which already indicates that it refers to the scorable_block_completion_handler.  So rather than showing up as (for example) `ScorableCompletionHandlerTestCase.test_scorable_block_completion_handler_submits_completion`, it will be `ScorableCompletionHandlerTestCase.test_handler_submits_completion`.